pkg/controller/bundle/bundle_unpacker.go

```diff
@@ -0,0 +1,388 @@
+	if err != nil {
+		return
+	}
+
+	var role *rbacv1.Role
+	role, err = c.ensureRole(cm.GetNamespace(), cm.GetName())
+	if err != nil {
+		return
+	}
+
+	_, err = c.ensureRoleBinding(role.GetNamespace())
+	if err != nil {
+		return
+	}
+
+	var job *batchv1.Job
+	job, err = c.ensureJob(cm.GetNamespace(), cm.GetName(), result.Path)
+	if err != nil {
+		return
+	}
+
+	cond := result.GetCondition(operatorsv1alpha1.BundleLookupPending)
+	now := c.now()
+	if job == nil || len(job.Status.Conditions) == 0 || len(job.Status.Conditions) > 0 && job.Status.Conditions[0].Type != batchv1.JobComplete {
+		if cond.Status != corev1.ConditionTrue {
+			cond.Status = corev1.ConditionTrue
+			cond.Reason = "JobIncomplete"
+			cond.Message = "unpack job not completed"
+			cond.LastTransitionTime = &now
+			result.SetCondition(cond)
+		}
+
+		return
+	}
+
+	result.bundle, err = c.loader.Load(cm)
+	if err != nil {
+		// TODO: This may never be used depending on how the returned error is interpreted; consider removing.
+		if cond.Status != corev1.ConditionTrue {
+			cond.Status = corev1.ConditionTrue
+			cond.Reason = "ConfigMapNotLoaded"
+			cond.Message = "unpack configmap was not loaded"
+			cond.LastTransitionTime = &now
+			result.SetCondition(cond)
+		}
+
+		return
+	}
+
+	// A successful load should remove the pending condition
+	result.RemoveCondition(operatorsv1alpha1.BundleLookupPending)
+
+	return
+}
+
+func (c *configmapUnpacker) ensureConfigmap(namespace, name string) (cm *corev1.ConfigMap, err error) {
+	fresh := &corev1.ConfigMap{}
+	fresh.SetNamespace(namespace)
+	fresh.SetName(name)
+
+	cm, err = c.cmLister.ConfigMaps(fresh.GetNamespace()).Get(fresh.GetName())
+	if apierrors.IsNotFound(err) {
+		cm, err = c.client.CoreV1().ConfigMaps(fresh.GetNamespace()).Create(fresh)
+	}
+
+	return
+}
+
+func (c *configmapUnpacker) ensureJob(configmapNamespace, configmapName, bundlePath string) (job *batchv1.Job, err error) {
+	fresh := c.job(configmapNamespace, configmapName, bundlePath)
+	job, err = c.jobLister.Jobs(fresh.GetNamespace()).Get(fresh.GetName())
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			job, err = c.client.BatchV1().Jobs(fresh.GetNamespace()).Create(fresh)
+		}
+
+		return
+	}
+
+	if equality.Semantic.DeepDerivative(fresh.Spec, job.Spec) {
+		return
+	}
+
+	// TODO: Decide when to fail-out instead of deleting the job
+	err = c.client.BatchV1().Jobs(job.GetNamespace()).Delete(job.GetName(), &metav1.DeleteOptions{})
+	job = nil
+	return
+}
+
+func (c *configmapUnpacker) ensureRole(configmapNamespace, configmapName string) (role *rbacv1.Role, err error) {
+	rule := rbacv1.PolicyRule{
+		APIGroups: []string{
+			"",
+		},
+		Verbs: []string{
+			"create", "get", "update",
+		},
+		Resources: []string{
+			"configmaps",
+		},
+		ResourceNames: []string{
+			configmapName,
+		},
+	}
+	fresh := &rbacv1.Role{
+		Rules: []rbacv1.PolicyRule{rule},
+	}
+	fresh.SetNamespace(configmapNamespace)
+	fresh.SetName(c.rbacName)
```

Comment:
since this uses the `rbacName` set for the bundle unpacker, won't this cause an issue for multiple bundle unpacks in the same namespace at the same time? Shouldn't the role name be based on the `configmapName` in some way (or generated with labels/ownerrefs?)

Reply:
Instead of generating a role per bundle, we can aggregate rules on a single role that gets updated. This should shake out to be eventually consistent in the event of contention between concurrent bundle. It's simple enough to switch it over -- I'll do that.

Reply:
Isn't this the rolebinding here?

Reply:
It should end up being a separate `PolicyRule` in the same `Role`.